libusb/os/darwin_usb.c

```diff
@@ -2332,27 +2306,36 @@ static int darwin_handle_transfer_completion (struct usbi_transfer *itransfer) {
   return usbi_handle_transfer_completion (itransfer, darwin_transfer_status (itransfer, tpriv->result));
 }
 
-#if !defined(HAVE_CLOCK_GETTIME)
 void usbi_get_monotonic_time(struct timespec *tp) {
-  mach_timespec_t sys_time;
+// If the SDK is new enough to have the clock_gettime() declaration, and the deplopment target is at least 10.12...
```

Comment:
> If the SDK is new enough to have the clock_gettime()
- isn't that what HAVE_CLOCK_GETTIME will tell us?

Reply:
They are similar, yes. But compile-time checks are generally preferable to configure-time checks. The latter are an anti-pattern IMNSHO, they are very cross-compile unfriendly.  Why go to the trouble of building a test program when you can just adjust your build with the preprocessor?  Especially when we need a solution that works both with autotools and Xcode.

Reply:
And feature checks are preferable to version checks :) but I agree this is not obvious when "cross-compiling" for a different macOS version. I guess configure is great for setting all the overall flags for a big build, but fine-navigation between versions subtleties are easier with the preprocessor.

Theoretically feature checks would be more robust e.g. in case this gets built in a pimped-up build environment where e.g. clock_gettime() could show up supported through compatibility add-ons (à la macports). But we can leave all that thinking to macports :)

Reply:
>And feature checks are preferable to version checks :)

The version check is the feature check.  clock_gettime() is documented to have been added in 10.12, thus checking the version is checking for the feature.

>but I agree this is not obvious when "cross-compiling" for a different macOS version

Or any cross-compilation.  Imagine trying to build libusb for iOS from macOS.  Or building for ARM on Intel.  Or building for FreeBSD from macOS. Any test you do on the machine that's building can give the wrong result for the machine you're building *for*.

But I don't want to start a religious argument. :)

Reply:
I am regularly cross-building libusb, e.g. for Windows on Linux. autoconf does the right thing if you've got the details right.